src/reducers-actions/orders.ts

```diff
@@ -22,13 +23,53 @@ export const appendOrders = (orders: AuctionElement[]): UpdateOrdersActionType =
   payload: { orders },
 })
 
+export const updateOrders = (orders: AuctionElement[]): UpdateOrdersActionType => ({
+  type: 'UPDATE_ORDERS',
+  payload: { orders },
+})
+
 export const updateOffset = (offset: number): UpdateOffsetActionType => ({
   type: 'UPDATE_OFFSET',
   payload: { offset },
 })
 
 export const INITIAL_ORDERS_STATE = { orders: [], offset: 0 }
 
+/**
+ * When orders are `deleted` from the contract, they are still returned, but with all fields set to zero.
+ * We will not display such orders.
+ *
+ * This function checks whether the order has been zeroed out.
+ * @param order The order object to check
+ */
+function isOrderDeleted(order: AuctionElement): boolean {
+  return (
+    order.buyTokenId === 0 &&
+    order.sellTokenId === 0 &&
+    order.priceDenominator.eq(ZERO) &&
+    order.priceNumerator.eq(ZERO) &&
+    order.validFrom === 0 &&
+    order.validUntil === 0
+  )
+}
+
+function classifyOrders(
+  allOrders: AuctionElement[],
+): { activeOrders: { [id: string]: number }; deletedOrdersIds: Set<string> } {
+  const activeOrders: { [id: string]: number } = {}
```

Comment:
why not also make this is a `Set`?

Reply:
I guess, you mean a Map, right?

Or ou mean that he wouldn't be storing the index. I think he did that to be more efficient updating. I also have some doubts if the optimization is worth it for the number of orders we expect. Arguably, you could search in a collection every time you want to update an order, and decide if it's addition/update. Anyways, I'm ok with the optimization too, and for sure is more efficient :)

Reply:
> why not also make this is a Set?

As Anxo noted, yeah, this is sort of a Map, but I'm using an Object instead. Guess I should go for a map?

Regardless, not a set because I need the index in the original array, and the order id is the key for quicker access.